src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java

```diff
@@ -429,12 +404,8 @@ else if (out1value instanceof String)
 			else if (out2value instanceof String)
 				setOutput(out2, (String)out2value);
 		}
-//AW:AFTER_TIMESLICES_END
 	}
 
-	/**
-	 * Required method returns a list of all input time series names.
-	 */
 	public String[] getInputNames()
```

Comment:
This method is defined in the super class, you can remove this and the array and just call this.getInputNames when needed, or cache it during beforeTimeSlices as appropriate.

Reply:
What do you think about caching in the super class;  the inputs, and outputs, should not be changing should they?

Reply:
We'll have to think about that. so long as it's an instance variable that should work without issue, I think.